core/unit_test/tools/include/ToolTestingUtilities.hpp

```diff
@@ -973,7 +973,7 @@ static uint64_t last_kernel_id;
 static uint32_t last_section_id;
 
 /** Subscribes to all of the requested callbacks */
-void set_tool_events_impl(const ToolValidatorConfiguration& config) {
+static void set_tool_events_impl(const ToolValidatorConfiguration& config) {
```

Comment:
Why `static` rather than `inline`?

Reply:
If we make this function `inline`, the compiler is free to choose to modify the `static` variable in any translation unit instead of the one we want.